src/bin/rdcore/kargs.rs

```diff
@@ -99,7 +99,7 @@ pub fn zipl(config: ZiplConfig) -> Result<()> {
         boot.mountpoint(),
         config.hostkey,
         config.rootfs,
-        config.kargs,
+        config.append_karg.as_ref().map(|v| v.join(" ")),
```

Comment:
This is not new, but for a follow-up: note that this function uses the kargs passed here in two ways: the first is in the SE path to create the sdboot blob in which it is appended to the regular kargs, the second is in the non-SE path to edit a temporary BLS entry in which we use `KargsEditor::append_if_missing`, which is not quite the same as the declared semantics of `--append-karg` (notice how for example `rdcore kargs` has both `--append` and `--append-if-missing`).

My suggestion is to:
1. pass in the `config.kargs` as a slice here
2. in the SE path, use `KargsEditor::append` when generating the blob
3. in the non-SE path, use `KargsEditor::append` for these kargs, and `append_if_missing` for `ignition.firstboot` only

Reply:
i'm not sure it's worth doing in this small PR, cause all of those look like a rework. Also in SE-path usage of `KargsEditor` would look like an overhead:
1) mirror `/boot/loader/entries` to `/tmp/smth`
2) append `kargs` to `tmp *.conf` 
3) read `kargs` for `genprotimg`

Reply:
`KargsEditor` operates directly on the string so doesn't require doing the `/tmp` hack.

Reply:
Ok, will re-work a bit `zipl` code in ongoing patches